spec/system/users/show_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+describe 'Users show page', type: :system do
+  describe 'checks the content of the users#show page' do
+    before(:each) do
+      @user3 = User.create(name: 'Hana', photo: 'https://livewiredemos.com/images/avatar.png', bio: 'Doctor from PEgypt.')
+      @post1 = Post.create!(title: 'Hello', text: 'here is my first post', author: @user3)
+      @post3 = Post.create!(title: 'Greetings', text: 'here is my third post.', author: @user3)
+      @post4 = Post.create!(title: 'fourth post', text: 'here is my fourth post.', author: @user3)
+      @post5 = Post.create!(title: 'fifth post', text: 'here is my fifth post.', author: @user3)
+    end
+
+    it "expects to show the user's profile picture" do
+      visit user_path(@user3)
+
+      expect(page).to have_css("img[src*='#{@user3.photo}']")
+    end
+
+    it "expects to see the user's username" do
+      visit user_path(@user3)
+
+      expect(page).to have_content(@user3.name)
+    end
+
+    it 'expects to see the number of posts the user has written' do
+      visit user_path(@user3)
+
+      expect(page).to have_content("Number of posts: #{@user3.posts.size}")
+    end
+
+    it "expects to see the user's bio" do
+      visit user_path(@user3)
+
+      expect(page).to have_content(@user3.bio)
+    end
+
+    it "expects to see the user's first 3 posts" do
+      visit user_path(@user3)
+
+      # expect(page).not_to have_content(@post1.title)
+      # expect(page).to have_content(@post3.title)
+      # expect(page).to have_content(@post4.title)
+      # expect(page).to have_content(@post5.title)
```

Comment:
- [ ] It would be better professionally if you remove any commented(unused) code to make your code more readable and clean. So please fix that.

Reply:
Ok Thanks Done

Reply:
by the way , I'm from Alexandria, too but I live in Kuwait. Nice to meet you here.